invoice/src/test/java/org/killbill/billing/invoice/usage/TestContiguousIntervalCapacityInArrear.java

```diff
@@ -243,6 +246,224 @@ public boolean apply(final InvoiceItem input) {
+        assertEquals(itemDetails1.size(), 2);
+        assertEquals(itemDetails1.get(0).getTier(), 1);
+        assertEquals(itemDetails1.get(0).getQuantity().intValue(), 99);
+        assertEquals(itemDetails1.get(0).getTierPrice().compareTo(BigDecimal.ONE), 0);
+        assertEquals(itemDetails1.get(1).getTier(), 1);
+        assertEquals(itemDetails1.get(1).getQuantity().intValue(), 5);
+        assertEquals(itemDetails1.get(1).getTierPrice().compareTo(BigDecimal.ONE), 0);
+
+        assertEquals(result.get(1).getAmount().compareTo(new BigDecimal("10.0")), 0, String.format("%s != 10.0", result.get(0).getAmount()));
+        assertEquals(result.get(1).getCurrency(), Currency.BTC);
+        assertEquals(result.get(1).getAccountId(), accountId);
+        assertEquals(result.get(1).getBundleId(), bundleId);
+        assertEquals(result.get(1).getSubscriptionId(), subscriptionId);
+        assertEquals(result.get(1).getPlanName(), planName);
+        assertEquals(result.get(1).getPhaseName(), phaseName);
+        assertEquals(result.get(1).getUsageName(), usage.getName());
+        assertTrue(result.get(1).getStartDate().compareTo(firstBCDDate) == 0);
+        assertTrue(result.get(1).getEndDate().compareTo(secondBCDDate) == 0);
+        // check item detail
+        List<UsageInArrearDetail> itemDetails2 = objectMapper.readValue(result.get(1).getItemDetails(), new TypeReference<List<UsageInArrearDetail>>() {});
+        assertEquals(itemDetails2.size(), 2);
+        assertEquals(itemDetails2.get(0).getTier(), 2);
+        assertEquals(itemDetails2.get(0).getQuantity().intValue(), 101);
+        assertEquals(itemDetails2.get(0).getTierPrice().compareTo(BigDecimal.TEN), 0);
+        assertEquals(itemDetails2.get(1).getTier(), 2);
+        assertEquals(itemDetails2.get(1).getQuantity().intValue(), 5);
+        assertEquals(itemDetails2.get(1).getTierPrice().compareTo(BigDecimal.TEN), 0);
+
+        assertEquals(result.get(2).getAmount().compareTo(new BigDecimal("100.0")), 0, String.format("%s != 100.0", result.get(0).getAmount()));
+        assertEquals(result.get(2).getCurrency(), Currency.BTC);
+        assertEquals(result.get(2).getAccountId(), accountId);
+        assertEquals(result.get(2).getBundleId(), bundleId);
+        assertEquals(result.get(2).getSubscriptionId(), subscriptionId);
+        assertEquals(result.get(2).getPlanName(), planName);
+        assertEquals(result.get(2).getPhaseName(), phaseName);
+        assertEquals(result.get(2).getUsageName(), usage.getName());
+        assertTrue(result.get(2).getStartDate().compareTo(secondBCDDate) == 0);
+        assertTrue(result.get(2).getEndDate().compareTo(endDate) == 0);
+        // check item detail
+        List<UsageInArrearDetail> itemDetails3 = objectMapper.readValue(result.get(2).getItemDetails(), new TypeReference<List<UsageInArrearDetail>>() {});
+        assertEquals(itemDetails3.size(), 2);
+        assertEquals(itemDetails3.get(0).getTier(), 3);
```

Comment:
Same remark here, should be tier `2`. 

Also could we add assertion on `tierUnit` (e.g `FOO` v.s `BAR`). Not only would be good for the test but to follow the test assertion -- not easy to understand if element `0` v.s `1` is `FOO` or `BAR`. And are we sure this is deterministic and all iterations of tests will produce same results?